### ISSUE TITLE
interaction with hyperref and destlabel=true

### ISSUE THREAD

**Issue by mrpiggi** [ho-tex/bookmark#1]
Following MWE raises the warning
`pdfTeX warning (dest): name{chapter.1} has been referenced but does not exist, replaced by a fixed one`
```latex
\documentclass{book}
\usepackage[destlabel=true]{hyperref}
\usepackage{bookmark}
\begin{document}
\chapter{Test}
\label{sec:chap}
\end{document}
```

**Comment by u-fischer** [comment]
yes, bookmark doesn't honor the destlabel setting. You try the following (but it is not much tested)

~~~~
\documentclass{book}
\usepackage[destlabel=true]{hyperref}
\usepackage{bookmark}

\makeatletter
\def\Hy@writebookmark#1#2#3#4#5{%
  \ifnum#4>\BKM@depth\relax
  \else
    \def\BKM@type{#5}%
    \ifx\BKM@type\Hy@bookmarkstype
      \begingroup
        \ifBKM@numbered
          \let\numberline\Hy@numberline
          \let\booknumberline\Hy@numberline
          \let\partnumberline\Hy@numberline
          \let\chapternumberline\Hy@numberline
        \else
          \let\numberline\@gobble
          \let\booknumberline\@gobble
          \let\partnumberline\@gobble
          \let\chapternumberline\@gobble
        \fi
        \bookmark[level=#4,dest={\HyperDestNameFilter{#3}}]{#2}%
      \endgroup
    \fi
  \fi
}
\makeatother
\begin{document}
\chapter{Test}
\label{sec:chap}
\chapter{Test2}
\label{sec:chap2}

\chapter{Test3}
\end{document}
~~~~

**Comment by mrpiggi** [comment]
This seems to work, at least I did not find any problems. Is this something, what would be integrated in the next version? Then I would wrap it within `\@ifpackagelater{bookmark}{2019/12/04}{}{...}`. Thanks for your help.

**Comment by u-fischer** [comment]
I added a new version in a test branch. But as there are no testfiles, it should be tried a bit before it is released to ctan.

**Comment by u-fischer** [comment]
the fix has been added to version 1.29 and uploaded to ctan 2020-11.
